### ISSUE TITLE
Just one column 'name' (and the many benefits from it)

### ISSUE THREAD

**Issue by ladlon** [turtletooth/tahoma2d#356]
This suggestion will hopefully address a few things...

Currently, columns have two names... the procedurally generated one (ex. Col16), and (if added) a user-defined name (ex. Left Leg).

The current function editor displays both, resulting in a kind of messy display... plus, you are always looking for the user-defined name anyway (since that was the whole point of giving it that custom name... so it's recognizable, rather than a generic Col name.

I would suggest, if possible, that there is just one name.  Each column starts with a procedural one (Col), but you can rename it to a custom one (ex. Tree) which replaces the procedural, rather than being a secondary name.

This would have the following benefits (I'd assume):

1) The function editor list (left side) would be much cleaner, plus it would be easier to spot the identifiable (user) names, rather than the non-descriptive Col ones.  With both currently being shown, the Col name only serves to make it harder to spot the user names, espeically since they are listed first (more visible, plus nicely lined up along the left margin), whereas the (more recognizable/useful) user name is tacked on at the end (not nicely lined up).

2) Expressions that reference columns would no longer be negatively affected by the column name being renamed as the result of the column being either moved in the column stack order, or by it being nudged/renamed as a result of a new column being introduced 'below' it.  Since the user name is not affected by the moving of a column, the current expression 'break' issue would (theoretically) not be an issue anymore.

3) Expressions themselves would be far more recognizable/logical, as they would reference recognizable names/objects, rather than generic Col names (ex. Car.x   vs  Col13.x)

**Comment by artisteacher** [comment]
You might want to add to the discussion on #114.

**Comment by DarrenTAnims** [comment]
Just to be clear. Columns only have 1 name. And levels have 1 name. When you create a level, you're naming the level. And if your current cell in the xsheet is on another drawing, then that level will appear in a new column that gets its own unique name.

But my  request (that I've made a lot of times for OpenToonz over the years) is that when you name a level, if it's going to be placed in a new column, then that column should take on the same initial name as the level.

And I think the function editor should only show the column name as it's the column that the animation values are applied to. You can move the drawings from different levels around, but the animation values affect whatever is on those frames in those columns.

**Comment by ladlon** [comment]
One of the things that repeatedly confused me in my first few days with OT/Tahoma2D is that when you click the + button to add a new level, you get a popup that is titled New Level... but it also creates a new column... so, as a beginner, it seemed like it was a button to create a new column (which, in doing so, would also come with a new level for you to start with).  That, plus the fact that there doesn't seem to be a 'Create New Column' button, kind of adds to the 'assumption' that the + button is to create a new column (which also then adds a blank level in it, as it figures you will want that anyway).

From the beginning, I understood columns vs levels... yet I actually started to doubt/question it over the next few days (partially because of the situation outlined above).  I mean, I 'get it'... but it was a source of confusion intially.

For a while, too, when I wanted to rename my columns (so they were more recognizable as far as what they contained), I woudl instinctively think that the top field of the New Level window was where you did it, and was briefly puzzled each time that it renamed the drawing group (level), rather than the column.  Not saying it was wrong... just a source of (some) confusion.

**Comment by ladlon** [comment]
I'm not talking about level names.  I'm talking about the fact that columns do have 2 names... the procedural (ex Col2), and the optional custom (ex. Left Arm).

**Comment by ladlon** [comment]
I do see your point about the column name being the one that should show in the function editor.  I agree, but I'm not suggesting the level name is used... but the column name... only that any custom names REPLACE the default 'col' name (rather than being appended (in brackets) to the col name in the list in the function editor.

**Comment by DarrenTAnims** [comment]
Yes. Now I understand. Thanks. The custom name should definitely replace the default col name. I don't know if anyone would use that default name if they've added a custom name.
